middleware/indexer.js

```diff
@@ -0,0 +1,23 @@
+const { Client } = require('pg');
+
+let client;
+async function getPgClient() {
+    if (!client) {
+        client = new Client({
+            connectionString: process.env.INDEXER_DB_CONNECTION,
+        });
+        await client.connect();
+    }
+
+    return client;
+}
+
+async function findAccountsByPublicKey(ctx) {
+    const { publicKey } = ctx.params;
+
+    const client = await getPgClient();
+    const { rows } = await client.query('SELECT account_id FROM access_keys WHERE public_key = $1', [publicKey]);
```

Comment:
All validation/sanitization of `publicKey` is handled by `pg` library?

Not sure where this is being used but maybe we need to throw "Invalid publicKey" 400 or 422?

Reply:
there is no sanitization needed, this is prepared SQL statement (i.e. you cannot just inject some stuff)